### PR TITLE
docs: set heading size CSS

### PR DESCRIPTION
**Pull request by nikkimk**
## What I did

1. set heading font-size in docs CSS

## Testing Instructions

From the [Card Guidelines in DP](https://deploy-preview-1112--red-hat-design-system.netlify.app/elements/card/guidelines/):
- [ ] Verify that heading `h1` is `48px`
- [ ] Verify that heading `h2` is `28px`
- [ ] Verify that heading `h3` is `24px`
- [ ] Verify that heading `h4` is `20px`

Use Element Inspector to temporarily change some headings to `h5` and `h6` locally in the browser:
- [ ] Verify that heading `h5` is `18px`
- [ ] Verify that heading `h6` is `16px`

## Notes to Reviewers

- Initial issue with recommended sizes based on  @coreyvickery's [comment in #1066](https://github.com/RedHat-UX/red-hat-design-system/pull/1066#issuecomment-1607936024)
- Font sizes based on @coreyvickery's [comment in this PR](https://github.com/RedHat-UX/red-hat-design-system/pull/1112#issuecomment-1619015116)
- Font sizes updated based on @coreyvickery's [latest comment](https://github.com/RedHat-UX/red-hat-design-system/pull/1112#issuecomment-1631391680)

### CODE REVIEW

**Comment by coreyvickery** [comment]
@nikkimk DP link is not working.

**Comment by zeroedin** [comment]
> @nikkimk DP link is not working.

@coreyvickery  here: https://deploy-preview-1112--red-hat-design-system.netlify.app/elements/card/guidelines/ fixed for you in link above

**Comment by coreyvickery** [comment]
@nikkimk Is this an issue?

https://ux.redhat.com/elements/accordion/

On Element and Pattern pages, the Accordion title text is spec'd to be an `h1` with the `--rh-font-size-heading-2xl` token (`3.0rem`, `48px`). Makes sense.

However, what is supposed to be the `h2` (Overview subheading text) is really an `h4` with the `--rh-font-size-heading-md` token (`1.75rem`, `28px`).

We are skipping `h2` and `h3` headings on Element and Pattern pages, but they are used on other pages.

If this is not an issue, then this is how `headings` map to tokens as outlined on the [tokens demo page](https://red-hat-design-tokens.netlify.app/#font-size-heading).

- `h1` = `--rh-font-size-heading-2xl` (`3.0rem`, `48px`)
- `h2` = `--rh-font-size-heading-xl` (`2.5rem`, `40px`)
- `h3` = `--rh-font-size-heading-lg` (`2.25rem`, `36px`)
- `h4` = `--rh-font-size-heading-md` (`1.75rem`, `28px`)
- `h5` = `--rh-font-size-heading-sm` (`1.5rem`, `24px`)
- `h6` = `--rh-font-size-heading-xs` (`1.25rem`, `20px`)

**Comment by nikkimk** [comment]
@coreyvickery made some updates. PTAL

**Comment by coreyvickery** [comment]
@bennypowers I’d still like to give Nikki feedback, please don’t close yet.

**Comment by bennypowers** [comment]
@coreyvickery no problem, just use the PR review process to either request changes or approve the PR

read all about it here: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews

**Comment by bennypowers** [comment]
@coreyvickery for example, if you want a specific change, click on `files changed`, go to the file in question, if you can find the line that needs to change, click on the `+` on that line and add your comment, and if not, click on the icon in that file's header and add a general comment to that file. those comments will be marked as pending

Then, when you're finished your review, click on `submit review` and add in some general comments, then select `request changes` and submit. from that point on, your explicit and formal approval will be necessary to dismiss your request for changes.

this is *the only way* to ensure that your requirements are addressed before a PR gets merged.

A more lenient alternative would just be to start a conversation on the PR by adding in a comment on any line or file in the `files changed` pane. Any comment thread started in that way will block merge until it's resolved, but anyone with write access can resolve a conversation, not just the person who started it.

just leaving a comment in the PR thread (like this comment i'm leaving here right now) will not prevent this pr from merging

File: docs/scss/_base/_base.scss
Comment (coreyvickery) [inline docs/scss/_base/_base.scss:1]: @nikkimk The sizes of `h5` and `h6` headings seem correct, but they are rendering `RedHatText` instead of `RedHatDisplay`. The `#` icons are also not aligned vertically.

`h5`
Screen Shot 2023-07-06 at 11 25 03 AM

`h6`
Screen Shot 2023-07-06 at 11 26 00 AM

Lastly, just making sure you saw [my comment](https://github.com/RedHat-UX/red-hat-design-system/pull/1112#issuecomment-1619015116) a few days ago about needing to skip `h2` and `h3` headings and if that is even possible. If what I wrote is confusing, ping me on chat.

**Review by coreyvickery — CHANGES_REQUESTED** [review]
Left comments about font weight and skipping heading sizes.

**Comment by nikkimk** [comment]
@coreyvickery if by skipping headings, you mean going from `h1` to h4, it is an accessibility issue.

**Comment by coreyvickery** [comment]
@nikkimk New values for ux dot ONLY as we discussed.

h1 = `48px`
h2 = `28px`
h3 = `24px`
h4 = `20px`
h5 = `18px`
h6 = `16px`

All headings have a line height of `1.3`.

h1 weight is `Regular`, all others are `Medium`.

**Review by coreyvickery — APPROVED** [review]

**Pull request merged**
